### PR TITLE
feat(cnpg): cnpg version tracking and imageselection

### PR DESCRIPTION
**Pull request by Ornias1993**
**Description**
We need to track cnpg major versions and add hooks to, eventually, allow major upgrades.
We also need to prevent accidental major version increases at all costs.

At the same time, this also requires us to explicitly start referencing containers, after all we need to track versions for said containers.

Fixes: #718
Fixes: #719

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**

**📃 Notes:**

Should enable continued work on:
https://github.com/truecharts/charts/issues/18737

**✔️ Checklist:**

- [ ] ⚖️ My code follows the style guidelines of this project
- [ ] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

### CODE REVIEW

File: library/common/templates/spawner/_cnpg.tpl [hunk 1]
```diff
@@ -43,6 +43,28 @@
       {{- $_ := set $.Values.configmap $recoveryConfigMapName $recConfig -}}
     {{- end -}}
 
+
+    {{/* Handle version string */}}
+    {{- $pgversion := $objectData.version | default $objectData.global.fallbackDefaults.pgversion -}}
```
Comment (Ornias1993) [inline library/common/templates/spawner/_cnpg.tpl:48]:
```suggestion
    {{- $pgversion := $objectData.version | default $.Values.global.fallbackDefaults.pgversion -}}
```

**Comment by Ornias1993** [comment]
TODO:
- tests
- modify renovate to correctly track versions and not bump major

File: library/common/templates/class/cnpg/_cluster.tpl [hunk 2]
```diff
@@ -131,6 +131,14 @@
   {{- with $objectData.cluster.walStorage.accessModes -}}
     {{- $walAccessModes = . -}}
   {{- end -}}
+  
+  {{/* Ensure version and container tracking */}}
+  {{- $imageKey := printf "%s%sImage" $objectData.type $objectData.pgversion }}
+  {{- $imageValue := (index $rootCtx.Values $imageKey) }}
```
Comment (stavros-k) [inline library/common/templates/class/cnpg/_cluster.tpl:137]: I  think we have a template that fetches image dict by just passing the $rootCtx and an imageSelector
Making all the validations needed!

Comment (Ornias1993) [reply library/common/templates/class/cnpg/_cluster.tpl:137]: good call, image and imageName use the same format I think?

Comment (Ornias1993) [reply library/common/templates/class/cnpg/_cluster.tpl]: For now those validations would be enough yes. 
We can later add more clear errors if we like.

**Comment by Ornias1993** [comment]
New TODO:

- Fixing bugs
- tests
- validation of type and version
- modify renovate to correctly track versions and not bump major

**Comment by Ornias1993** [comment]
New TODO:

- tests
- modify renovate to correctly track versions and not bump major

**Pull request merged**